Hardcode screensize for WebKit Internals.

megangardner opened the pull request:
#### f3c3c38858ff5b47d7c9afa2cd8008cae0f8a416
<pre>
Hardcode screensize for WebKit Internals.
<a href="https://bugs.webkit.org/show_bug.cgi?id=243112">https://bugs.webkit.org/show_bug.cgi?id=243112</a>

Reviewed by Tim Horton.

We need to save off the minimumEffectiveDeviceWidth so that does not interfere with this different
hardcoded value.
We also need to save off the original size so that it can be restored when exiting fullscreen.

* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(WebKit::WKWebViewState::applyTo):
(WebKit::WKWebViewState::store):
(-[WKFullScreenWindowController enterFullScreen]):
(-[WKFullScreenWindowController beganEnterFullScreenWithInitialFrame:finalFrame:]):
(-[WKFullScreenWindowController beganExitFullScreenWithInitialFrame:finalFrame:]):

Canonical link: <a href="https://commits.webkit.org/252757@main">https://commits.webkit.org/252757@main</a>
</pre>

hortont424 commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
```diff
@@ -134,6 +135,7 @@ void applyTo(WKWebView* webView)
         webView.scrollView.minimumZoomScale = _savedMinimumZoomScale;
         webView.scrollView.maximumZoomScale = _savedMaximumZoomScale;
         webView.scrollView.bouncesZoom = _savedBouncesZoom;
+        webView._minimumEffectiveDeviceWidth = _savedMinimumEffectiveDeviceWidth;
```
Changelog should explain this

hortont424 commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
```diff
@@ -532,11 +535,15 @@ - (void)enterFullScreen
 
     _fullScreenState = WebKit::WaitingToEnterFullScreen;
     _blocksReturnToFullscreenFromPictureInPicture = manager->blocksReturnToFullscreenFromPictureInPicture();
+    _originalWindowSize = [webView window].frame.size;
```
Changelog should explain this move

hortont424 commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
```diff
@@ -532,11 +535,15 @@ - (void)enterFullScreen
     [_window setWindowLevel:UIWindowLevelNormal - 1];
     [_window setHidden:NO];
+#if HAVE(UIKIT_WEBKIT_INTERNALS)
+    [_window setFrame:CGRectMake(0, 0, 960, 540)];
```
Possibly this should come from a constant or something? It's weird that it's just hardcoded, though I get that that was the idea.

megangardner committed:
Hardcode screensize for WebKit Internals.
https://bugs.webkit.org/show_bug.cgi?id=243112

Reviewed by Tim Horton.

We need to save off the minimumEffectiveDeviceWidth so that does not interfere with this different
hardcoded value.
We also need to save off the original size so that it can be restored when exiting fullscreen.

* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(WebKit::WKWebViewState::applyTo):
(WebKit::WKWebViewState::store):
(-[WKFullScreenWindowController enterFullScreen]):
(-[WKFullScreenWindowController beganEnterFullScreenWithInitialFrame:finalFrame:]):
(-[WKFullScreenWindowController beganExitFullScreenWithInitialFrame:finalFrame:]):

Canonical link: https://commits.webkit.org/252757@main
